### ISSUE TITLE
does adding a student after a piece is already assigned to a course create the assignments for that student?

### ISSUE THREAD

**Issue by nwself** [JMU-CIME/CPR-Music-Backend#17]

**Comment by nwself** [comment]
It does not. I would prefer that adding a student not do this magic on its own. Can frontend call like `/api/courses/:slug/reassign` with student_id in params? Is that too much hassle on frontend?

@hcientist

**Comment by hcientist** [comment]
So, sorry just to check: you’re saying don’t automatically do this just because a request came in to add the student but rather have a special endpoint that would look at all currently assigned pieces for the course? Or should the front end enumerate them and then make all of those calls? @nwself

**Comment by nwself** [comment]
Yes, I'm saying don't automatically do this. I'll leave it to you to decide whether adding a special endpoint vs frontend making one call per assignment is better. Possibly frontend enumerating is best because teacher might not want to assign every previous assignment if student is added late or something? 

But let me know if I should make this special endpoint

**Comment by hcientist** [comment]
I think the special endpoint would be good. It seems like there's little problem if we get it wrong and the student never does some old assignment or does it and the teacher never bothers marking it, while if the teacher doesn't notice to add the old assignments to the student, they may not realize the student is missing them until the student reports it. Like the fact that assignments can be given to individual students is a v3 thing. Right now the UI for teachers is to choose a piece, so it seems weird to put it into this part of the frontend which might not even have the relevant bits from the db at the time of adding a single student (i guess likewise for adding in bulk from roster... 🤔)

I guess it presents at least 1 question: what should be the due date of assignments created in this way 😕 because we're lying to the user about this right now, it's weird. like I guess we have to pull the due date of the first assignment to another student enrollment in the same course (and since there's no UI for individual due dates rn, this will be correct)

**Comment by nwself** [comment]
Agreed. Due date should either be like you say: a due date from any existing assignment, or something arbitrary like either N days from today or course end date. Benefit of something arbitrary is that you won't accidentally come up with a due date that is already passed which in theory will flag the submission as late even thought the student didn't have any options

**Comment by hcientist** [comment]
If I have late added a student I think it wouldn’t surprise me if something was late. At the same time I’m wondering if the more likely case is that just 1 or small number of students might be late added even before deadlines just because of beginning of term shuffles? So I was thinking the likely path not even having any due yet, but I have no data about this. Also rn there’s no “late” status in the interface, but ofc we can imagine there will be eventually…

it seems to me that the likelihood of the arbitrary due date ever being right is extremely low, and the likelihood of copying it is unknown, but at least in the (possibly nonexistent) use case I imagined it would be the correct due date (if copied from another in their course)

**Comment by hcientist** [comment]
> But let me know if I should make this special endpoint

Please do!
